src/st-cuts.c

```diff
@@ -1240,27 +1232,28 @@ int igraph_i_all_st_mincuts_pivot(const igraph_t *graph,
   nomin=igraph_vector_size(&M);
   for (i=0; i<nomin; i++) {
     long int min=(long int) VECTOR(Sbar_invmap)[ (long int) VECTOR(M)[i] ];
-    if (!igraph_estack_iselement(T, min)) { break; }
+    if (min != target)
+      if (!igraph_estack_iselement(T, min)) { break; }
   }
   if (i!=nomin) {
     /* OK, we found a pivot element. I(S,v) contains all elements
        that can reach the pivot element */
     igraph_vector_t Isv_min;
-    long int isvlen;
     IGRAPH_VECTOR_INIT_FINALLY(&Isv_min, 0);
     *v=(long int) VECTOR(Sbar_invmap)[ (long int) VECTOR(M)[i] ];
     /* TODO: restricted == keep ? */
     IGRAPH_CHECK(igraph_bfs(graph, /*root=*/ (igraph_integer_t) *v,/*roots=*/ 0,
-			    /*mode=*/ IGRAPH_IN, /*unreachable=*/ 0,
-			    /*restricted=*/ &keep, /*order=*/ &Isv_min,
-			    /*rank=*/ 0, /*father=*/ 0, /*pred=*/ 0,
-			    /*succ=*/ 0, /*dist=*/ 0, /*callback=*/ 0,
-			    /*extra=*/ 0));
-    for (isvlen=0; isvlen<no_of_nodes; isvlen++) {
-      if (!IGRAPH_FINITE(VECTOR(Isv_min)[isvlen])) { break; }
+          /*mode=*/ IGRAPH_IN, /*unreachable=*/ 0,
+          /*restricted=*/ &keep, /*order=*/ &Isv_min,
+          /*rank=*/ 0, /*father=*/ 0, /*pred=*/ 0,
+          /*succ=*/ 0, /*dist=*/ 0, /*callback=*/ 0,
+          /*extra=*/ 0));
+    for (j=0; j<no_of_nodes; j++) {
+      igraph_real_t u = VECTOR(Isv_min)[j];
+      if (!IGRAPH_FINITE(u)) { break; }
+      if (!igraph_estack_iselement(T, u))
+        igraph_vector_push_back(Isv, u);
```

Comment:
I don't quite follow whre `Isv` originally came from, but unless it is guaranteed that this `push_back` will never need to allocate more storage for it, there should be an `IGRAPH_CHECK` on this.

Reply:
Thanks!